### PR TITLE
Integration with api

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: provider/http.go [hunk 1]
```diff
@@ -14,15 +14,15 @@ import (
 type HTTPClient struct {
 	pb.UnimplementedApiServer
 
-	baseURL    string
-	httpClient *http.Client
-	requestID  utils.RequestID
-	privateKey *solana.PrivateKey
-	authHeader string
+	BaseURL    string
```
Comment (aspin) [inline provider/http.go]: reasoning behind this? I don't see a need to expose any of these attributes outside of this package based on this PR

File: provider/http.go [hunk 2]
```diff
@@ -14,15 +14,15 @@ import (
...
+	HttpClient *http.Client
+	RequestID  utils.RequestID
+	PrivateKey *solana.PrivateKey
+	AuthHeader string
 }
 
 func (h *HTTPClient) GetAuthHeader() string {
```
Comment (aspin) [inline provider/http.go]: nit: should be called `AuthHeader()`, or can be deleted entirely if we need to make the properties public

File: provider/http.go (hunk 1 above)
Comment (aspin) [reply provider/http.go]: ah, for integration tests? Does the constructor now provide good enough configurability?

Comment (tinystarinagalaxy) [reply provider/http.go]: oh, these are left overs, will revert these.

File: provider/http.go [hunk 3]
```diff
@@ -49,25 +49,36 @@ func NewHTTPLocal() *HTTPClient {
 	return NewHTTPClientWithOpts(nil, opts)
 }
 
+// NewHTTPCustomClient
+func NewHTTPCustomClient(authHeader, endpoint string) *HTTPClient {
```
Comment (tinystarinagalaxy) [inline provider/http.go:49]: we use this in solana-trader api, don't need the props to be public

File: provider/http.go (hunk 1 above)
Comment (tinystarinagalaxy) [reply provider/http.go]: reverted

**Review by aspin — APPROVED** [review]

File: provider/http.go (hunk 2 above)
Comment (tinystarinagalaxy) [reply provider/http.go]: removed this func and made all props private

**Pull request merged**